src/knex.js

```diff
@@ -0,0 +1,83 @@
+import Raw from './raw';
+import Client from './client';
+
+import makeKnex from './util/make-knex';
+import parseConnection from './util/parse-connection';
+import { SUPPORTED_CLIENTS, CLIENT_ALIASES } from './constants';
+
+export default function Knex(config) {
+  // If config is string, try to parse it
```

Comment:
added comment with explanation for each block, please verify the accuracy of them